delfin/drivers/dell_emc/power_store/power_store.py

```diff
@@ -0,0 +1,167 @@
+        pass
+
+    @staticmethod
+    def parse_alert(context, alert):
+        return rest_handler.RestHandler.get_parse_alerts(alert)
+
+    def get_alert_sources(self, context):
+        return self.rest_handler.get_alert_sources(self.storage_id)
+
+    def list_controllers(self, context):
+        return self.rest_handler.get_controllers(self.storage_id)
+
+    def list_disks(self, context):
+        return self.rest_handler.get_disks(self.storage_id)
+
+    def list_ports(self, context):
+        hardware_d = self.rest_handler.get_port_hardware()
+        fc_ports = self.rest_handler.get_fc_ports(self.storage_id, hardware_d)
+        fc_ports.extend(
+            self.rest_handler.get_eth_ports(self.storage_id, hardware_d))
+        fc_ports.extend(
+            self.rest_handler.get_sas_ports(self.storage_id, hardware_d))
+        return fc_ports
+
+    def reset_connection(self, context, **kwargs):
+        self.rest_handler.logout()
+        self.rest_handler.login()
+
+    def add_trap_config(self, context, trap_config):
+        pass
+
+    def remove_trap_config(self, context, trap_config):
+        pass
+
+    @staticmethod
+    def get_access_url():
+        return 'https://{ip}'
+
+    def collect_perf_metrics(self, context, storage_id, resource_metrics,
+                             start_time, end_time):
+        LOG.info('The system(storage_id: %s) starts to collect storage and'
+                 ' volume performance, start_time: %s, end_time: %s',
+                 storage_id, start_time, end_time)
+        metrics = []
+        if resource_metrics.get(constants.ResourceType.STORAGE):
+            storage_metrics = self.rest_handler.get_storage_metrics(
+                storage_id,
+                resource_metrics.get(constants.ResourceType.STORAGE),
+                start_time, end_time,
+                constants.ResourceType.STORAGE)
+            metrics.extend(storage_metrics)
+            LOG.info('The system(storage_id: %s) stop to collect storage'
+                     ' performance, The length is: %s',
+                     storage_id, len(storage_metrics))
+        if resource_metrics.get(constants.ResourceType.STORAGE_POOL):
+            pool_metrics = self.rest_handler.get_pool_metrics(
+                storage_id,
+                resource_metrics.get(constants.ResourceType.STORAGE_POOL),
+                start_time, end_time,
+                constants.ResourceType.STORAGE_POOL)
+            metrics.extend(pool_metrics)
+            LOG.info('The system(storage_id: %s) stop to collect pool'
+                     ' performance, The length is: %s',
+                     storage_id, len(pool_metrics))
+        if resource_metrics.get(constants.ResourceType.VOLUME):
+            volume_metrics = self.rest_handler.get_volume_metrics(
+                storage_id,
+                resource_metrics.get(constants.ResourceType.VOLUME),
+                start_time, end_time,
+                constants.ResourceType.VOLUME)
```

Comment:
According to the function name `get_volume_metrics` , we already know which kind of performance it can collect, so the parameter `constants.ResourceType.VOLUME` is can be deleted, and it can be used directly inside the function, so as other functions.

Reply:
It has been modified.